.github/workflows/test_minimal_deps.yml

```diff
@@ -0,0 +1,39 @@
+name: Minimal Dependency Tests
+
+on:
+  push:
+    paths-ignore:
+      - 'docs/**'
+  pull_request:
+    paths-ignore:
+      - 'docs/**'
+  workflow_dispatch:
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: ["3.7"]
+
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python }}
+    - name: Cache Python Dependencies
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.pythonLocation }}
+        key: minimal-env-${{ runner.os }}-python-${{ matrix.python }}-${{ hashFiles('requirements/minimal_requirements.txt') }}
+        restore-keys: |
+          minimal-env-${{ runner.os }}-python-${{ matrix.python }}-
```

Comment:
Could this not accidentally use a cache that has different minimal requirements than in `minimal_requirements.txt`?

Reply:
If we update the requirements.txt file we would always get a new hash and therefore don't use the cache.

The only complication that can happen here is that if we have two branches and one of them introduces a requirements change it might update the cache and then the second branch might use the newly cached dependencies. The question is if we think that this is a likely scenario that might break things.

Reply:
I thought the `restore-keys` are alternative keys that can be used if there is a cache miss. Maybe I didn't understand correctly. But I would be very conservative with caching. Accidentally using wrong dependencies might be difficult to spot.